README.md

```diff
@@ -1,20 +1,506 @@
 # JavaScript Objects
 
-## Objectives
-+ Explain what an object is in JavaScript is
-+ Create an object in JS
-+ Access a value from an object
-+ Add a key-value pair to an object
-+ Delete a key-value pair from an object
-+ Iterate over key-value pairs in an object
-
-
-## What the Student Already Knows
-
-- Variables
-- Data Type- Strings and Integers
-- Functions
-- Booleans, Comparison Operators, Flow Control 
-- Arrays
+## Overview
+JavaScript is an "object-oriented" language. Objects in one form or another are
+an important part of every modern programming language. In this lesson we will
+explore why they are so useful them and how we can use them.
+
+
+## What is an object?
+An object in JavaScript is fundamentally a way to store key-value pairs. Huh?
+What does that mean? Let's look at the following example:
+
+A common problem when writing a chat server is to keep track of connected
+users. Typically each user has an unique name (= "username") and an open
+connection to the server (= "socket"). A server needs to be able to keep track
+of all connected clients. One way using a data structure you already know is
+using an array of username-socket pairs:
+
+```js
+const connections = [['max', conn0], ['alex', conn1], ['james', conn2]];
+```
+
+This totally works: In order to connect a client, you would `push` a new
+`['username', socket]` pair to the `connections` array. In order to disconnect
+a client, you would `splice` the respective pair from the array.
```

Comment:
I like this way of introducing students to objects using something that they've just learned about.